### PR TITLE
Mediator pattern

### PR DESCRIPTION
**Pull request by ArvApu**
Added ability to chat. Implemented using mediator design pattern. To sue press C.

### CODE REVIEW

**Comment by JustasGau** [comment]
Atrodo viskas tvarkingai

File: DonjinKrawler-Server/src/main/java/donjinkrawler/chatroom/ChatRoom.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package donjinkrawler.chatroom;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ChatRoom implements ChatMediator {
```
Comment (JustasGau) [inline DonjinKrawler-Server/src/main/java/donjinkrawler/chatroom/ChatRoom.java]: Turėtų būt final

File: DonjinKrawler-Common/src/main/java/krawlercommon/enemies/big/BigZombie.java [hunk 2]
```diff
@@ -38,6 +38,6 @@ public void setStrategies() {
 
     @Override
     public void debug() {
-        System.out.println("Big zombie:" + this.getID() + " has been created");
+        // ystem.out.println("Big zombie:" + this.getID() + " has been created");
```
Comment (EdotJ) [inline DonjinKrawler-Common/src/main/java/krawlercommon/enemies/big/BigZombie.java:41]: Why is this commented out ?

File: DonjinKrawler-Common/src/main/java/krawlercommon/enemies/small/SmallZombie.java [hunk 3]
```diff
@@ -38,6 +38,6 @@ public void setStrategies() {
 
     @Override
     public void debug() {
-        System.out.println("Small zombie:" + this.getID() + " has been created");
+        // System.out.println("Small zombie:" + this.getID() + " has been created");
```
Comment (EdotJ) [inline DonjinKrawler-Common/src/main/java/krawlercommon/enemies/small/SmallZombie.java:41]: Ditto

File: DonjinKrawler-Server/src/main/java/donjinkrawler/chatroom/ChatRoom.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @Override
+    public RoomMate getRoomMate(int id) {
+        return this.roomMates.get(id);
+    }
+
+    @Override
+    public void sendMessage(RoomMate sender, String message) {
+        for(RoomMate receiver : this.roomMates.values()) {
+            // Message should not be received by the user sending it
+            if(receiver != sender){
```
Comment (EdotJ) [inline DonjinKrawler-Server/src/main/java/donjinkrawler/chatroom/ChatRoom.java]: Formatting

File: DonjinKrawler-Server/src/main/java/donjinkrawler/chatroom/ChatRoom.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package donjinkrawler.chatroom;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public final class ChatRoom implements ChatMediator {
+
+    private final Map<Integer, RoomMate> roomMates;
+
+    public ChatRoom(){
```
Comment (EdotJ) [inline DonjinKrawler-Server/src/main/java/donjinkrawler/chatroom/ChatRoom.java]: Formatting

**Review by EdotJ — APPROVED** [review]

**Pull request merged**